### ISSUE TITLE
Error compiling sample custom rule

### ISSUE THREAD

**Issue by karsunke** [palantir/tslint#2303]
### Bug Report

- __TSLint version__: 4.5.0, 4.5.1
- __TypeScript version__: 2.1.5, 2.2.1
- __Running TSLint via__: CLI 

#### TypeScript code being linted

```ts
import * as ts from "typescript";
import { Rules, RuleFailure, RuleWalker } from "tslint";

export class Rule extends Rules.AbstractRule {
    public static FAILURE_STRING = "import statement forbidden";

    public apply(sourceFile: ts.SourceFile): RuleFailure[] {
        return this.applyWithWalker(new NoImportsWalker(sourceFile, this.getOptions()));
    }
}

// The walker takes care of all the work.
class NoImportsWalker extends RuleWalker {
    public visitImportDeclaration(node: ts.ImportDeclaration) {
        // create a failure at the current position
        this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING));

        // call the base version of this visitor to actually parse this node
        super.visitImportDeclaration(node);
    }
}
```

#### Actual behavior
When I try to compile the sampleRule via CLI, i get the following errors:
```
../node_modules/tslint/lib/language/utils.d.ts(40,82): error TS2694: Namespace 'ts' has no exported member 'ModifierFlags'.
../node_modules/tslint/lib/language/utils.d.ts(53,84): error TS2694: Namespace 'ts' has no exported member 'ObjectFlags'.
../node_modules/tslint/lib/language/utils.d.ts(98,48): error TS2694: Namespace 'ts' has no exported member 'BinaryOperatorToken'.
../node_modules/tslint/lib/language/walker/syntaxWalker.d.ts(74,44): error TS2694: Namespace 'ts' has no exported member 'NumericLiteral'.
```

#### Expected behavior
It should compile without an error.

**Comment by karsunke** [comment]
Ok I now tried it with a newer nodeJS version (v7.7.1) and it worked. I tried it with v6.9.5 and it did not work. Maybe you should add this to the documentation of the custom rule setup.

**Comment by chrismbarr** [comment]
The docs say to use Node 7.x: https://github.com/palantir/tslint#development

**Comment by karsunke** [comment]
Yes, but only in the "Development" part. Not in the "custom rule" part.

**Comment by chrismbarr** [comment]
but... you have to develop a custom rule... that's what development is...

😕

**Comment by nchen63** [comment]
That error doesn't appear related to node, though. ModifierFlags was added in 2.1, so maybe your Typescript version wasn't updated
